commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -208,4 +220,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * Builds URL with image coords for MediaWiki API calls
+     * Example URL: https://commons.wikimedia.org/w/api.php?action=query&prop=categories|coordinates|pageprops&format=json&clshow=!hidden&coprop=type|name|dim|country|region|globe&codistancefrompoint=38.11386944444445|13.356263888888888&
+     * generator=geosearch&redirects=&ggscoord=38.11386944444445|13.356263888888888&ggsradius=100&ggslimit=10&ggsnamespace=6&ggsprop=type|name|dim|country|region|globe&ggsprimary=all&formatversion=2
+     */
+    public static class UrlBuilder {
+        private static String buildUrl (String coords){
+
+            Uri.Builder builder = Uri.parse("https://commons.wikimedia.org/").buildUpon();
+
+            builder.appendPath("w")
+                    .appendPath("api.php")
+                    .appendQueryParameter("action", "query")
+                    .appendQueryParameter("prop", "categories|coordinates|pageprops")
+                    .appendQueryParameter("format", "json")
+                    .appendQueryParameter("clshow", "!hidden")
+                    .appendQueryParameter("coprop", "type|name|dim|country|region|globe")
+                    .appendQueryParameter("codistancefrompoint", coords)
+                    .appendQueryParameter("generator", "geosearch")
+                    .appendQueryParameter("ggscoord", coords)
+                    .appendQueryParameter("ggsradius", "100")
+                    .appendQueryParameter("ggslimit", "10")
+                    .appendQueryParameter("ggsnamespace", "6")
+                    .appendQueryParameter("ggsprop", "type|name|dim|country|region|globe")
+                    .appendQueryParameter("ggsprimary", "all")
+                    .appendQueryParameter("formatversion", "2");
+
+            return builder.toString();
+        }
+    }
+
 }
```

Comment:
Hi @Niedzielski,  thanks for the feedback. :) I am traveling for these 3
days without access to my work computer, but I will see to these as soon as
I get back.  Cheers!
On 31/12/2015 11:01 am, "Stephen Niedzielski" <notifications@github.com>
wrote:

> In commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java
> <https://github.com/nicolas-raoul/apps-android-commons/pull/33#discussion_r48634701>
> :
>
> > +                    .appendQueryParameter("clshow", "!hidden")
> > +                    .appendQueryParameter("coprop", "type|name|dim|country|region|globe")
> > +                    .appendQueryParameter("codistancefrompoint", coords)
> > +                    .appendQueryParameter("generator", "geosearch")
> > +                    .appendQueryParameter("ggscoord", coords)
> > +                    .appendQueryParameter("ggsradius", "100")
> > +                    .appendQueryParameter("ggslimit", "10")
> > +                    .appendQueryParameter("ggsnamespace", "6")
> > +                    .appendQueryParameter("ggsprop", "type|name|dim|country|region|globe")
> > +                    .appendQueryParameter("ggsprimary", "all")
> > +                    .appendQueryParameter("formatversion", "2");
> > +
> > +            return builder.toString();
> > +        }
> > +    }
> > +
> >  }
>
> I think I repro'd some of these weird Maven / pom issues :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nicolas-raoul/apps-android-commons/pull/33/files#r48634701>
> .
>